cpp/src/cider/type/plan/BinaryExpr.cpp

```diff
@@ -275,9 +277,8 @@ JITExprValue& BinOper::codegenFixedSizeLogicalFun(CodegenContext& context,
         auto lhs_data = lhs_val.getValue();
         auto rhs_null = rhs_val.getNull();
         auto rhs_data = rhs_val.getValue();
-        auto null = (lhs_null && rhs_null) || (lhs_null && rhs_data && !rhs_null) ||
-                    (rhs_null && lhs_data && !lhs_null);
-        return set_expr_value(null, lhs_data && rhs_data);
+        auto null = (lhs_null & rhs_null) | (lhs_null & rhs_data) | (rhs_null & lhs_data);
```

Comment:
seems you change the result.  should `lhs_null & rhs_data & !rhs_null`

Reply:
Unnecessary, for ```lhs_null && rhs_data && !rhs_null``` and ```rhs_null && lhs_data && !lhs_null```, ```lhs_null``` and ```rhs_null``` are always reversed. In ```rhs_null && lhs_data && !lhs_null```, if ```rhs_null``` is true, ```!lhs_null``` will be true and could be omitted. If ```rhs_null``` is false, then ```rhs_null && lhs_data && !lhs_null``` will be false and could be omitted in the entire ```OR``` expression.

Reply:
Seems reasonable.  why @imajiayu found a bug?

Reply:
The original code is `(lhs_null && rhs_null) || (lhs_null && lhs_data) || (rhs_null && rhs_data);`, there are minor differences with this PR. I fixed it and @bigPYJ1151 simplified it now.

Reply:
got it